src/router/mod.rs

```diff
@@ -70,25 +83,77 @@ where
             aliases: Mutex::new(HashMap::new()),
             subscribers: Mutex::new(HashMap::new()),
             filters: RwLock::new(HashMap::new()),
-            conversation_relays: Mutex::new(HashMap::new()),
+            relay_nodes: Mutex::new(HashMap::new()),
+            global_relay_node: Mutex::new(RelayNode::new()),
         }
     }
 
-    /// Get all conversations that are subscribed to a specific relay.
-    pub async fn get_conversations_by_relay(
-        &self,
-        relay: String,
-    ) -> Result<Vec<String>, ConversationError> {
-        let conversations = self.conversation_relays.lock().await;
-        let mut result = Vec::new();
-
-        for (id, relays) in conversations.iter() {
-            if relays.contains(&relay) {
-                result.push(id.clone());
+    pub async fn add_relay(&self, url: String) -> Result<(), <C as Channel>::Error> {
+        self.channel().add_relay(url.clone()).await?;
+
+        let mut relay_nodes = self.relay_nodes.lock().await;
+
+        relay_nodes.insert(
+            url.clone(),
+            RelayNode {
+                conversations: HashSet::new(),
+            },
+        );
```

Comment:
Here you should check if the relay is already present to avoid overwriting it:

```suggestion
        relay_nodes.entry(url.clone())
            .or_insert_with(|| {
                RelayNode {
                    conversations: HashSet::new(),
                }
            });
```